### ISSUE TITLE
Album Artwork Not Updating

### ISSUE THREAD

**Issue by singhpaul** [Pe8er/Playbox.widget#47]
Quite often, the album artwork does not update when a new album begins playing.

There is no debug info in the Console and it would appear not to be an issue with Ubersicht not updating the widget because the progress bar and the meta info do update. 

Hitting "Refresh All Widgets" does resolve the problem temporarily.

Any thoughts? Happy to help debug.

**Comment by singhpaul** [comment]
This appears to be a caching issue. 

When a new album plays, the artwork is saved with filename in **cover#.ext** format, where # is a random integer between 0 and 9.  

If a file already exists with that name, it's simply overwritten, but the "browser" instance continues to display the old cached version of the file, even though it has been updated in the file system. 

A solution that is working for me is to add a timestamp URL parameter to the end of the covert art URL which forces a refresh by changing the following line:

`set currentCoverURL to getPathItem(currentCoverURL)`

to

`set currentCoverURL to (getPathItem(currentCoverURL)) & ("?ts=") & (do shell script "date +%s")`

This issue was less likely to manifest in earlier versions of the plugin because the random number was selected between 0 and 999, but that ultimately led to GBs of permanent disk space used by 2,000 artwork files (because of .jpg and .png formats). Should note that this issue was, at least in part, "resolved" by a PR that I submitted.
